> hiding output except if something fails, in which case all output gets dumped

That sounds like a good solution to me, updated the PR. Full output if a unit test fails (in this case, I manually changed an assertion):

```
●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●●
199 successes / 1 failure / 0 errors / 0 pending : 0.0 seconds

Failure → /home/florian/Projects/neovim/test/unit/os/fs_spec.lua @ 76
fs function os_dirname returns OK and writes current directory into the buffer if it is large
    enough
/home/florian/Projects/neovim/test/unit/os/fs_spec.lua:77: Expected objects to be the same.
Passed in:
(number) 1
Expected:
(number) 0

stack traceback:
	/home/florian/Projects/neovim/test/unit/os/fs_spec.lua:77: in function </home/florian/Projects/neovim/test/unit/os/fs_spec.lua:76>

-- Output to stderr:
E94: No matching buffer for _test_
E94: No matching buffer for _test_
E94: No matching buffer for _test_
E94: No matching buffer for test
CMake Error at /home/florian/Projects/neovim/cmake/RunTests.cmake:25 (message):
  Running unit tests failed with error: 1.
```

--- a/cmake/RunTests.cmake
+++ b/cmake/RunTests.cmake
@@ -17,8 +17,10 @@ execute_process(
   COMMAND ${BUSTED_PRG} -v -o ${BUSTED_OUTPUT_TYPE}
     --lpath=${BUILD_DIR}/?.lua ${TEST_PATH}
   WORKING_DIRECTORY ${WORKING_DIR}
+  ERROR_VARIABLE err
   RESULT_VARIABLE res)
 
 if(NOT res EQUAL 0)
+  message(STATUS "Output to stderr:\n${err}")
   message(FATAL_ERROR "Running ${TEST_TYPE} tests failed with error: ${res}.")
 endif()

